k-distribution/src/main/scripts/lib/pyk/ktool.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+
+import os
+import subprocess
+
+from .kastManip import *
+
+class KPrint:
+    """Given a kompiled directory, build an unparser for it.
+    """
+
+    def __init__(self, kompiledDirectory):
+        self.kompiledDirectory = kompiledDirectory
+        self.definition        = readKastTerm(self.kompiledDirectory + '/compiled.json')
+        self.symbolTable       = buildSymbolTable(self.definition, opinionated = True)
+
+    def prettyPrint(self, kast):
+        """Given a KAST term, pretty-print it using the current definition.
+
+        -   Input: KAST term in JSON.
+        -   Output: Best-effort pretty-printed representation of the KAST term.
+        """
+        return prettyPrintKast(kast, self.symbolTable)
+
+class KProve(KPrint):
+    """Given a kompiled directory and a main file name, build a prover for it.
+    """
+
+    def __init__(self, kompiledDirectory, mainFileName, useDirectory = None):
+        super(KProve, self).__init__(kompiledDirectory)
+        self.directory      = '/'.join(self.kompiledDirectory.split('/')[0:-1])
```

Comment:
You should use a library to get the parent directory here. What happens when you have `/my/path/test-kompiled/.`
Also, `directory` -> `parentDir`? I find it more descriptive.

Reply:
I'm now using the `Path` object from `pathlib` for this. It does make it nicer.